File: src/brs/db/store/AliasStore.java
package brs.db.store;

import brs.Alias;
import brs.db.SignumKey;
import brs.db.VersionedEntityTable;

import java.util.Collection;

public interface AliasStore {
  SignumKey.LongKeyFactory<Alias> getAliasDbKeyFactory();
  SignumKey.LongKeyFactory<Alias.Offer> getOfferDbKeyFactory();

  VersionedEntityTable<Alias> getAliasTable();

  VersionedEntityTable<Alias.Offer> getOfferTable();

  Collection<Alias> getAliasesByOwner(long accountId, String name, Long tld, int from, int to);

  Collection<Alias> getTLDs(int from, int to);
  
  Collection<Alias.Offer> getAliasOffers(long account, long buyer, int from, int to);

  Alias getAlias(String aliasName, long tld);
  
  Alias getTLD(String tldName);
  
  Alias getTLD(long tldId);
}

File: src/main/java/brs/db/store/AliasStore.kt
package brs.db.store

import brs.Alias
import brs.db.BurstKey
import brs.db.VersionedEntityTable

interface AliasStore {
    val aliasDbKeyFactory: BurstKey.LongKeyFactory<Alias>
    val offerDbKeyFactory: BurstKey.LongKeyFactory<Alias.Offer>

    val aliasTable: VersionedEntityTable<Alias>

    val offerTable: VersionedEntityTable<Alias.Offer>

    fun getAliasesByOwner(accountId: Long, from: Int, to: Int): Collection<Alias>

    fun getAlias(aliasName: String): Alias
}
